libs/template/reader.go

```diff
@@ -0,0 +1,155 @@
+package template
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/databricks/cli/libs/cmdio"
+)
+
+type Reader interface {
+	// FS returns a file system that contains the template
+	// definition files. This function is NOT thread safe.
+	FS(ctx context.Context) (fs.FS, error)
+
+	// Close releases any resources associated with the reader
+	// like cleaning up temporary directories.
+	Close() error
```

Comment:
`Close` pairs with `Open`. Per the comment it sounds like `Cleanup` is a better name.